tests/integration/test_session.py

```diff
@@ -19,3 +21,27 @@ def _two_factor_auth():
                                         match_requests_on=match):
             r = self.session.get('https://api.github.com/users/sigmavirus24')
             assert r.status_code == 200
+
+    @pytest.fixture(autouse=True)
+    def test_token_auth_with_netrc_works(self, tmpdir):
+
+        self.token_login()
+
+        cassette_name = self.cassette_name('token_auth_with_netrc')
+
+        netrc = tmpdir.join(".netrc")
+        netrc.write(
+            "machine api.github.com\n"
+            "login sigmavirus24\n"
+            "password invalid_token_for_test_verification"
+        )
+
+        with mock.patch.dict('os.environ', {'HOME': tmpdir}):
```

Comment:
So to test this behaviour I think I need the above tmpdir fixture, but enabling that seems to cause this class to lose it's `self.gh` attribute resulting in an error being thrown on `self.token_login()` above.